lwm2m: uninitialized variable in this function

**Describe the bug**
With Network, LwM2M and by default CONFIG_LWM2M_RW_JSON_SUPPORT=y in Zephyr 2.5 this warning show up:

` /zephyrproject/zephyr/subsys/net/lib/lwm2m/lwm2m_rw_json.c:147:16: warning: 'c' may be used uninitialized in this function [-Wmaybe-uninitialized]
  147 |  uint8_t cont, c;`

**Environment (please complete the following information):**
 - OS: Linux,
 - Toolchain: Zephyr SDK

--- a/subsys/net/lib/lwm2m/lwm2m_rw_json.c
+++ b/subsys/net/lib/lwm2m/lwm2m_rw_json.c
@@ -144,7 +144,7 @@ static void json_add_char(struct lwm2m_input_context *in,
 static int json_next_token(struct lwm2m_input_context *in,
 			   struct json_in_formatter_data *fd)
 {
-	uint8_t cont, c;
+	uint8_t cont, c = 0;
 	bool escape = false;
 
 	(void)memset(fd, 0, sizeof(*fd));

